extension/app/ts/simulation/services/SimulationModeEthereumClientService.ts

```diff
@@ -529,6 +529,26 @@ export class SimulationModeEthereumClientService {
 		return await this.ethereumClientService.getTransactionByHash(hash)
 	}
 
+	public readonly getTokenDecimals = async (token: bigint) => {
+		const tokenInterface = new ethers.Interface(['function decimals() view returns (uint8)'])
+		const balanceOfCallData = stringToUint8Array(tokenInterface.encodeFunctionData('decimals'))
+		const callTransaction = {
+			type: '1559' as const,
+			from: MOCK_ADDRESS,
+			to: token,
+			input: balanceOfCallData,
+			maxFeePerGas: 0n,
+			maxPriorityFeePerGas: 0n,
+			gas: 21000n,
+			value: 200000000000000000000000n,
+			accessList: [],
+			nonce: 0n,
+			chainId: await this.getChainId()
+		}
```

Comment:
Several of these should be optional and don't really make sense for `call`s.  We should also use `as const` to help the type checker along (just as a general best practice when defining constant objects).
```suggestion
		const callTransaction = {
			type: '1559',
			from: MOCK_ADDRESS,
			to: token,
			input: balanceOfCallData,
			value: 200000000000000000000000n,
		} as const
```

Reply:
We are not actually making a call here, but a multicall, in which case we need to convert the call to a transaction. I have done that in a way that call requires a transaction directly to have the same API

Reply:
hmm actually I can just Pick here